File: tests/e2e/rate_limit.py
import asyncio
from playwright.async_api import async_playwright

async def run_test():
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        page = await browser.new_page()
        
        print("--- Iniciando Teste E2E: Rate Limiting ---")
        try:
            await page.goto("http://localhost:8080/admin-login")

            for i in range(1, 7):
                await page.fill('input[placeholder="usuário"]', "admin")
                await page.fill('input[placeholder="••••••••"]', "senha_errada")
                await page.click('button[type="submit"]')
                content = await page.content()
                if "bloqueado" in content.lower():
                    print(f"Tentativa {i}: Bloqueado OK")
                    break
                print(f"Tentativa {i}: Erro comum OK")

            await page.click('button:has-text("Esqueci minha senha admin")')
            for i in range(1, 5):
                await page.fill('input[placeholder="usuário"]', "inexistente")
                await page.click('button:has-text("Verificar Usuário")')
                content = await page.content()
                if "bloqueado" in content.lower():
                    print(f"Recuperação {i}: Bloqueado OK")
                    break
        except Exception as e:
            print(f"Falha no teste: {e}")
            exit(1)
        finally:
            await browser.close()

if __name__ == "__main__":
    asyncio.run(run_test())

File: tests/e2e/rate_limit.mjs
import { chromium } from 'playwright';

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  
  console.log('--- Iniciando Teste E2E: Rate Limiting ---');
  
  try {
    await page.goto('http://localhost:8080/admin-login');

    for (let i = 1; i <= 6; i++) {
      await page.fill('input[placeholder="usuário"]', 'admin');
      await page.fill('input[placeholder="••••••••"]', 'senha_errada');
      await page.click('button[type="submit"]');
      const msg = await page.textContent('.auth-form');
      if (msg.includes('bloqueado')) {
        console.log(`Tentativa ${i}: Bloqueado OK`);
        break;
      }
      console.log(`Tentativa ${i}: Erro comum OK`);
    }

    await page.click('button:has-text("Esqueci minha senha admin")');
    for (let i = 1; i <= 4; i++) {
      await page.fill('input[placeholder="usuário"]', 'inexistente');
      await page.click('button:has-text("Verificar Usuário")');
      const msg = await page.textContent('.auth-form');
      if (msg.includes('bloqueado')) {
        console.log(`Recuperação ${i}: Bloqueado OK`);
        break;
      }
    }

  } catch (e) {
    console.error('Falha no teste:', e);
    process.exit(1);
  } finally {
    await browser.close();
  }
})();
